src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -53,6 +57,70 @@ export const KIND_CALL_TRANSFER = "call_transfer";
 const INITIAL_ROOMS_SHOWN = 3; // Number of rooms to show at first
 const INCREMENT_ROOMS_SHOWN = 5; // Number of rooms to add when 'show more' is clicked
 
+/**
+ * Iterate backwards through the message history, returning messages that are
+ * visible to all members of the room (`m.room.history_visibility` is
+ * `world_readable` or `shared`), until the most recent message that is not.
+ *
+ * This function is intended to be used with MatrixClient.shareKeysForMessages.
+ */
+function iterateShareableHistoryForRoom(client, room) {
```

Comment:
Would this function make more sense in the JS SDK...? It doesn't really seem to have a UI component at least...